dask_cuda/explicit_comms/dataframe/shuffle.py

```diff
@@ -526,23 +529,32 @@ def shuffle(
     # TODO: can we do this without using `submit()` to avoid the overhead
     #       of creating a Future for each dataframe partition?
 
-    dsk = {}
+    futures = []
     for rank in ranks:
         for part_id in rank_to_out_part_ids[rank]:
-            dsk[(name, part_id)] = c.client.submit(
-                getitem,
-                shuffle_result[rank],
-                part_id,
-                workers=[c.worker_addresses[rank]],
+            futures.append(
+                c.client.submit(
+                    getitem,
+                    shuffle_result[rank],
+                    part_id,
+                    workers=[c.worker_addresses[rank]],
+                )
             )
 
     # Create a distributed Dataframe from all the pieces
-    divs = [None] * (len(dsk) + 1)
-    ret = new_dd_object(dsk, name, df_meta, divs).persist()
+    divs = [None] * (len(futures) + 1)
+    kwargs = {"meta": df_meta, "divisions": divs, "prefix": "explicit-comms-shuffle"}
+    try:
+        ret = dd.from_delayed(futures, **kwargs).persist()
+    except TypeError:
+        # This version of dask may not support `prefix`
+        # See: https://github.com/dask/dask-expr/pull/991
+        kwargs.pop("prefix")
+        ret = dd.from_delayed(futures, **kwargs).persist()
```

Comment:
991 is now merged, so dask-expr>1.0.4 will support `prefix`.

Reply:
It should be "okay" to leave out the prefix for older versions of dask-expr: When query-planning is enabled, we are skipping the tests where we assert that the collection name starts with "explicit-comms-shuffle". That is because those tests are checking that the "explicit-comms" monkey-patching logic is working - which we do not support when query-planning is enabled anyway.